polymer/src/sample-list.html

```diff
@@ -0,0 +1,191 @@
+<link rel="import" href="../node_modules/@npm-polymer/polymer/polymer-element.html">
+<link rel="import" href="../node_modules/@npm-polymer/app-route/app-route.html">
+<link rel="import" href="../node_modules/@npm-polymer/iron-flex-layout/iron-flex-layout.html">
+<link rel="import" href="sample-category-data.html">
+<link rel="import" href="sample-list-item.html">
+
+<dom-module id="sample-list">
+
+  <template>
+
+    <style>
+
+      .grid {
+        @apply --layout-horizontal;
+        @apply --layout-wrap;
+        @apply --layout-justified;
+        margin: 0 10px 32px 10px;
+        padding: 0;
+        list-style: none;
+      }
+
+      .grid li {
+        -webkit-flex: 1 1;
+        flex: 1 1;
+        -webkit-flex-basis: 33%;
+        flex-basis: 33%;
+        max-width: 33%;
+      }
+
+      .grid a {
+        display:block;
+        text-decoration: none;
+      }
+
+      @media (max-width: 767px) {
+        .hero-image {
+          display: none;
+        }
+
+        .grid  li {
+          -webkit-flex-basis: 50%;
+          flex-basis: 50%;
+          max-width: 50%;
+        }
+      }
+
+    </style>
+
+    <!--
+      app-route provides the name of the category.
+    -->
+    <app-route
+        route="[[route]]"
+        pattern="/:category"
+        data="{{routeData}}"></app-route>
+
+    <!--
+      sample-category-data provides the category data for a given category name.
+    -->
+    <sample-category-data
+        id="categoryData"
+        category-name="[[routeData.category]]"
+        category="{{category}}"
+        failure="{{failure}}"></sample-category-data>
+
+    <header>
+      <h1>[[category.id]]</h1>
+    </header>
+
+    <ul class="grid" hidden$="[[failure]]">
+      <dom-repeat items="[[_getListItems(category.items)]]" initial-count="4">
+        <template>
+          <li>
+            <a href$="[[_getItemHref(item)]]"><sample-list-item item="[[item]]"></sample-list-item></a>
+          </li>
+        </template>
+      </dom-repeat>
+    </ul>
+
+    <!--
+      shop-network-warning shows a warning message when the items can't be rendered due
+      to network conditions.
+    -->
+    <shop-network-warning
+        hidden$="[[!failure]]"
+        offline="[[offline]]"
+        on-try-reconnect="_tryReconnect"></shop-network-warning>
+
+  </template>
+
+  <script>
+
+    class SampleList extends Polymer.Element {
+
+      static get is() { return 'sample-list'; }
+
+      static get properties() { return {
+
+        category: Object,
+
+        route: Object,
+
+        routeData: Object,
+
+        visible: {
+          type: Boolean,
+          value: false
+        },
+
+        offline: {
+          type: Boolean,
+          observer: '_offlineChanged'
+        },
+
+        failure: Boolean
+
+      }}
+
+      static get observers() { return [
+        '_categoryChanged(category, visible)'
+      ]}
+
+      connectedCallback() {
+        super.connectedCallback();
+        this.isAttached = true;
```

Comment:
I think Polymer does this by default already.

Reply:
I took it from [here](https://github.com/Polymer/shop/blob/master/src/shop-list.html)

Reply:
Oh, apparently it is only on the legacy mixin. I missed that in the upgrade guide. Never mind this comment then :smile: